frontend/src/js/components/subscription/SubscriptionPage.tsx

```diff
@@ -0,0 +1,393 @@
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { ChangeEvent, ReactNode, useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import { Alert, Button, FormControl, FormControlLabel, FormHelperText, Radio, RadioGroup, TextField, Typography } from '@mui/material';
+
+import { SupportLink } from '@northern.tech/common-ui/SupportLink';
+import { ADDONS, Addon, AvailableAddon, AvailablePlans, PLANS, Plan } from '@northern.tech/store/appSlice/constants';
+import { getStripeKey } from '@northern.tech/store/appSlice/selectors';
+import { TIMEOUTS } from '@northern.tech/store/commonConstants';
+import { getDeviceLimit } from '@northern.tech/store/devicesSlice/selectors';
+import { getHasCurrentPricing, getOrganization } from '@northern.tech/store/organizationSlice/selectors';
+import { getBillingPreview, getCurrentCard, getUserBilling, getUserSubscription, requestPlanChange } from '@northern.tech/store/organizationSlice/thunks';
+import { useAppDispatch } from '@northern.tech/store/store';
+import { useDebounce } from '@northern.tech/utils/debouncehook';
+import { Elements } from '@stripe/react-stripe-js';
+
+import { addOnsToString } from '../settings/Upgrade';
+import { SubscriptionAddon } from './SubscriptionAddon';
+import { SubscriptionDrawer } from './SubscriptionDrawer';
+import { SubscriptionSummary } from './SubscriptionSummary';
+
+let stripePromise = null;
+export type PreviewPrice = { addons: { [key in AvailableAddon]: number }; plan: number; total: number };
+
+const DIVISIBILITY_STEP = 50;
+const enterpriseDeviceCount = PLANS.enterprise.minimalDeviceCount;
+const planOrder = Object.keys(PLANS);
+const enterpriseRequestPlaceholder = 'Tell us a little about your requirements and device fleet size, so we can provide you with an accurate quote';
+type SelectedAddons = { [key in AvailableAddon]: boolean };
+
+type ContactReason = 'reduceLimit' | 'overLimit';
+interface ContactReasonProps {
+  reason: ContactReason;
+}
+const ContactReasonAlert = ({ reason }: ContactReasonProps) => {
+  const reasonMap: { [key in ContactReason]: ReactNode | string } = {
+    reduceLimit: `For over ${enterpriseDeviceCount} devices, please contact for pricing.`,
+    overLimit: (
+      <div>
+        If you want to reduce your device limit, please contact <SupportLink variant="email " />
+      </div>
+    )
+  };
+  return (
+    <Alert severity="info" className="margin-bottom-x-small margin-top-x-small">
+      {reasonMap[reason]}
+    </Alert>
+  );
+};
```

Comment:
or as an alternative to have the reason map tied to the notification:
```suggestion
const contactReasons = {
  reduceLimit: {
    id: 'reduceLimit',
    alert: (
      <div>
        For over {enterpriseDeviceCount} devices, please contact <SupportLink variant="email" /> for pricing.
      </div>
    )
  },
  overLimit: {
    id: 'overLimit',
    alert: (
      <div>
        If you want to reduce your device limit, please contact <SupportLink variant="email" />.
      </div>
    )
  }
} as const;
interface ContactReasonProps {
  reason: keyof typeof contactReasons;
}
const ContactReasonAlert = ({ reason }: ContactReasonProps) => (
  <Alert severity="info" className="margin-bottom-x-small margin-top-x-small">
    {contactReasons[reason].alert}
  </Alert>
);
```
+ adjusting the `setContactReason` type + calls?